### ISSUE TITLE
Should world_axis_physical_types report the axis types for each pixel axis?

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/ndcube#240]
### Description

Currently ```NDCube.world_axis_physical_types``` (and ```NDCubeSequence.world_axis_physical_types``` and ```NDCubeSequence.cube_like_world_axis_physical_types```) simply return all the physical axis types represented in the WCS.  It does not say which pixel axis corresponds to which physical axis, even though in many cases, a naive one-to-one mapping will give the intuitively correct answer.

It would be highly useful if ```world_axis_physical_types``` or another property were to explicitly provide this mapping, both in the pixel -> physical and the physical -> pixel directions.

### Current Behavior
Say the NDCube's first two dimesions correspond longitude and latitude (i.e. they are dependent) and the third corresponds to an  independent type, e.g. wavelength.

```python
>>> len(NDCube.dimensions)  # See the number of pixel dimensions in the NDCube
3
>>> NDCube.world_axis_physical_types
('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat', 'em.wl')
```

Although this simply returns all physical axes, a naive reading of it suggests that the first axis is longitude, the second is latitude, and the third is wavelength.  However in truth, the first two are both longitude and latitude and the 3rd is wavelength.

### Desired Behavior
```python
>>> NDCube.world_to_pixel_axis_names()  # Ordering based on output of NDCube.world_axis_physical_types
((0, 1), (0, 1), 2)
>>> NDCube.pixel_to_world_axis_names()
(('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat'), ('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat'), 'em.wl')
```
Perhaps these functions could have a similar API to ```NDCube.axis_world_coords()``` and take strings and axis indices to return only the values the user is interested in.
```python
>>> NDCube.world_to_pixel_axis_names('lon')
(0, 1)
>>> NDCube.pixel_to_world_axis_names(0)
('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat')
```

These methods could not only search the WCS, but also the ```extra_coords```.

**Comment by DanRyanIrish** [comment]
@Cadair, this is something we discussed quite a while ago but put on the long finger.  Now [sunraster issue #131](https://github.com/sunpy/sunraster/issues/131) has brought this up again.  We should discuss whether the above feature addition is useful/feasible/timely.

**Comment by DanRyanIrish** [comment]
This is the same issue as #220

**Comment by DanRyanIrish** [comment]
Addressed by #281
